Update k8s auth docs for new parameter (#9992)
hashicorp/vault: website/pages/api-docs/auth/kubernetes/index.mdx

public key used to validate the JWT signature and the necessary information to
access the Kubernetes API.

> If they aren't specified, Vault will use the local CA cert and service account

Reviewer: ```suggestion
By default Vault will use the local CA cert and service account
```

JWT for the `kubernetes_ca_cert` and `token_reviewer_jwt` parameters,
respectively, when running in a Kubernetes pod. This behavior may be disabled by

> setting `disable_local_ca_jwt` to `true`.

Reviewer: If users set this to true do they then also need to specify `kubernetes_ca_cert` and `token_reviewer_jwt` as well?

Author: If they set `disable_local_ca_jwt` to `true`, then either `kubernetes_ca_cert` or `pem_keys` needs to be specified as well. `token_reviewer_jwt` is optional.

Author: I'm not sure if it helped, but I tried to elaborate a little more in 7fe90f0.

- `issuer` `(string: "")` - Optional JWT issuer. If no issuer is specified, then this plugin will
  use `kubernetes.io/serviceaccount` as the default issuer.
- `disable_iss_validation` `(bool: false)` - Disable JWT issuer validation. Allows to skip ISS validation.
- `disable_local_ca_jwt` `(bool: false)` - Disable defaulting to the local CA cert and service account JWT when running in a Kubernetes pod.

### Caveats

> Either `kubernetes_ca_cert` or `pem_keys` must be set.

Reviewer 2: This is dependent on the value from `disable_local_ca_jwt`?

Author: Good question, tried to clarify in 6588f33 and e6d098f.
